### PR TITLE
feat(users): Implementing user addition, edition and deletion

### PR DESCRIPTION
**Pull request by german-mergestat**
Implementing User `Addition`, `Edition` and `Deletion`.

- Unit Testing

Add User:

add-user

Edit User:

edit-user

Remove User:

remove-user

### CODE REVIEW

File: ui/src/api-logic/graphql/mutations/add-users.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { gql } from '@apollo/client'
```
Comment (patrickdevivo) [inline ui/src/api-logic/graphql/mutations/add-users.ts:1]: let's rename this file `manage-users.ts` since it's not just for adding users

Comment (german-mergestat) [reply ui/src/api-logic/graphql/mutations/add-users.ts]: Done [7441bbe](https://github.com/mergestat/mergestat/pull/572/commits/7441bbe80abf9aa84a1bf2c078fc6c21f869e668)

File: ui/src/views/settings/modals/add-user-modal/index.tsx [hunk 2]
```diff
@@ -57,29 +70,53 @@ export const AddUserModal: React.FC = () => {
...
+              <Input value={username} placeholder="username"
+                data-testid={TEST_IDS.usersAddUsername}
+                onChange={(e: ChangeEvent<HTMLInputElement>) => setUsername(e.target.value)}
+                onKeyPress={(e) => (e.key === 'Enter' && handleAddUser())}
+              />
             </div>
             <div>
               <Label>Password</Label>
-              <Input type='password' />
+              <Input type='password' value={password} placeholder="username"
```
Comment (patrickdevivo) [inline ui/src/views/settings/modals/add-user-modal/index.tsx:81]:
```suggestion
              <Input type='password' value={password} placeholder="password"
```

File: ui/src/views/settings/modals/add-user-modal/index.tsx [hunk 3]
```diff
@@ -57,29 +70,53 @@ export const AddUserModal: React.FC = () => {
...
+                data-testid={TEST_IDS.usersAddPassword}
+                variant={error ? 'error' : 'default'}
+                onChange={(e: ChangeEvent<HTMLInputElement>) => setPassword(e.target.value)}
+                onKeyPress={(e) => (e.key === 'Enter' && handleAddUser())}
+              />
             </div>
             <div>
               <Label>Confirm password</Label>
-              <Input type='password' />
+              <Input type='password' value={passwordConfirm} placeholder="username"
```
Comment (patrickdevivo) [inline ui/src/views/settings/modals/add-user-modal/index.tsx:90]:
```suggestion
              <Input type='password' value={passwordConfirm} placeholder="confirm password"
```

File: ui/src/views/settings/modals/edit-user-modal/index.tsx [hunk 4]
```diff
@@ -1,37 +1,60 @@
...
+  const handleUpdateUser = () => {
+    setError(false)
+
+    // If password or confirm password are typed
+    if (password || passwordConfirm) {
+      if (password !== passwordConfirm) {
+        setError(true)
+      } else { // Update password and role as well
+        updatePassword({ variables: { username: usernameEdit, password } })
+        updateRole({ variables: { username: usernameEdit, role } })
```
Comment (patrickdevivo) [inline ui/src/views/settings/modals/edit-user-modal/index.tsx:49]: I think we should only call this `updateRole` if there is a role change - we should be able to detect this yeah? We can avoid the extra API request if only the password is being changed.

File: ui/src/@types.tsx [hunk 5]
```diff
@@ -6,7 +6,9 @@ export type AlertType = 'warning' | 'info' | 'success' | 'error'
 
 export type RepoType = 'github' | 'gitlab' | 'bitbucket' | 'other'
 
-export type UserType = 'Admin' | 'User' | 'Read-Only' | 'Unknown'
+export type UserTypeUI = 'Admin' | 'User' | 'Read-Only' | 'Unknown'
```
Comment (patrickdevivo) [inline ui/src/@types.tsx:9]:
```suggestion
export type UserTypeUI = 'Admin' | 'User' | 'Read Only' | 'Unknown'
```

File: ui/src/utils/constants.ts [hunk 6]
```diff
@@ -84,9 +97,33 @@ export enum EXPORT_FORMAT {
...
+  ADMIN = 'ADMIN',
+  USER = 'USER',
+  READ_ONLY = 'READ_ONLY'
+}
+
+export const USER_ROLES = [
+  {
+    key: 'ADMIN',
+    name: 'Admin',
+    desc: 'Admins have access to everything'
```
Comment (patrickdevivo) [inline ui/src/utils/constants.ts:117]:
```suggestion
    desc: 'Admins can take any action'
```

File: ui/src/utils/constants.ts [hunk 7]
```diff
@@ -84,9 +97,33 @@ export enum EXPORT_FORMAT {
...
+  },
+  {
+    key: 'USER',
+    name: 'User',
+    desc: 'Users can create, read and update data, but can’t delete any data '
```
Comment (patrickdevivo) [inline ui/src/utils/constants.ts:122]:
```suggestion
    desc: 'Users can take most actions, but cannot create new users'
```

**Review by patrickdevivo — APPROVED** [review]

**Pull request merged**